src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -753,15 +876,44 @@ private void HandleExternalReplication(DatabaseRecord newRecord, List<IDisposabl
 
         private void SetExternalReplicationProperties(DatabaseRecord newRecord, List<ExternalReplicationBase> externalReplications)
         {
-            foreach (var externalReplication in externalReplications)
+            for (var i = 0; i < externalReplications.Count; i++)
             {
+                var externalReplication = externalReplications[i];
                 if (ValidateConnectionString(newRecord, externalReplication, out var connectionString) == false)
                 {
                     continue;
                 }
 
                 externalReplication.Database = connectionString.Database;
                 externalReplication.ConnectionString = connectionString;
+                
+                if (externalReplication is PullReplicationAsSink sink && 
+                    sink.Mode == (PullReplicationMode.Incoming | PullReplicationMode.Outgoing))
+                {
+                    // we have dual mode here, need to split it
+                    sink.Mode = PullReplicationMode.Incoming;
+                    
+                    var other = new PullReplicationAsSink
+                    {
+                        Database = sink.Database,
+                        Disabled = sink.Disabled,
+                        Incoming = sink.Incoming,
+                        Mode = PullReplicationMode.Outgoing,
+                        Name = sink.Name,
+                        Url = sink.Url,
+                        ConnectionString = sink.ConnectionString,
+                        CertificatePassword = sink.CertificatePassword,
+                        Outgoing = sink.Outgoing,
+                        MentorNode = sink.MentorNode,
+                        TaskId = sink.TaskId,
+                        ConnectionStringName = sink.ConnectionStringName,
+                        HubDefinitionName = sink.HubDefinitionName,
+                        CertificateWithPrivateKey = sink.CertificateWithPrivateKey
+                    };
+
+                    i += 1;
```

Comment:
Is this correct? you are modifying while iterating and changing the `i`.
you probably will skip some `externalReplication` at the end.

Reply:
Yes, that is intentional, note that we are _adding_ an item to the list, the `i+=1` is the skip the one we just added

Reply:
but the ` externalReplications.Count` in the for loop is already fixed.. and you are inserting that item in the middle, so you will not go over the last item

Reply:
`i < externalReplications.Count`  is evaluated every single loop run